### ISSUE TITLE
Unpad fails for strings longer than the padding block size.

### ISSUE THREAD

**Issue by BernardIgiri** [richkzad/go-pkcs7#1]
Your unpad method incorrectly uses the block size to get the offset into the padded array, when it should instead use the length of the array -1. This causes errors whenever the content length exceeds the padding block size.

**Comment by richkzad** [comment]
You're right! Apparently I only ever used the Pad function when I wrote this.
